src/packet_types.c

```diff
@@ -183,16 +183,46 @@ void angular_velocity_block_init(AngularVelocityBlock *b, const uint32_t measure
 void acceleration_data_block_init(AccelerationDataBlock *b, const uint32_t measurement_time,
                                   const int8_t full_scale_range, const int16_t x_axis, const int16_t y_axis,
                                   const int16_t z_axis) {
-    memcpy(b->bytes, &measurement_time, sizeof(measurement_time));
-    memcpy(b->bytes + sizeof(measurement_time), &full_scale_range, sizeof(full_scale_range));
-    // One byte of dead space after FSR
-    memcpy(b->bytes + 1 + sizeof(measurement_time) + sizeof(full_scale_range), &x_axis, sizeof(x_axis));
-    memcpy(b->bytes + 1 + sizeof(measurement_time) + sizeof(full_scale_range) + sizeof(x_axis), &y_axis,
-           sizeof(y_axis));
-    memcpy(b->bytes + 1 + sizeof(measurement_time) + sizeof(full_scale_range) + sizeof(x_axis) + sizeof(z_axis),
-           &z_axis, sizeof(z_axis));
+    
 }
 
+/**
+ * Initializes a telemetry data block with the provided information.
+ * @param b The telemetry request block to be initialized.
+ * @param data_subtype_1 The first requested data type.
+ * @param used_1 A boolean to determine if the first data request is valid.
+ * @param data_subtype_2 The second requested data type.
+ * @param used_2 A boolean to determine if the second data request is valid.
+ * @param data_subtype_3 The third requested data type.
+ * @param used_3 A boolean to determine if the third data request is valid.
+ * @param data_subtype_4 The fourth requested data type.
+ * @param used_4 A boolean to determine if the fourth data request is valid.
+ * */
+void telemetry_request_block_init(TelemetryRequestBlock *b, const uint8_t data_subtype_1, const uint8_t used_1,
+                                  const uint8_t data_subtype_2, const uint8_t used_2, const uint8_t data_subtype_3,
+                                  const uint8_t used_3, const uint8_t data_subtype_4, const uint8_t used_4) {
+
+    b->bytes[0] |= data_subtype_1 << 2;
+    b->bytes[0] |= used_1;
+    b->bytes[1] |= data_subtype_2 << 2;
+    b->bytes[1] |= used_2;
+    b->bytes[2] |= data_subtype_3 << 2;
+    b->bytes[2] |= used_3;
```

Comment:
This is good, but to be really clean, instead of `used_3` do `(used_3 & 0x01)` (this ensures we only OR in the last bit).

Repeat this for all 4 bytes.